### DISCUSSION TITLE
I am not seeing the controls I am using being rendered as in the Uno Material documentation.

### DISCUSSION THREAD

**Discussion by mobynet1** [unoplatform/uno#17791] [Q&A] [answered]
I have scoured the documentation looking to see if I am missing something, but I am not seeing the controls being rendered as in [Uno Material](https://platform.uno/docs/articles/external/uno.themes/doc/material-getting-started.html?tabs=wizard%2Csingleproj#using-c-markup). When I use a textbox for instance, it looks nothing like the text box in the diagram at the top of the page.

I know this next screenshot is very grainy, but this is what I am expecting:

![image](https://github.com/user-attachments/assets/4e4f3f23-a454-42b9-bc61-f2ffcb032b98)

This is what I am getting:

![image](https://github.com/user-attachments/assets/26ff83e8-033e-4a04-ad88-e75b16257e85)

I have applied so styling so that it looks almost identical to the Telerik Maui textbox, but it makes me wonder if I am even actually using the Uno Material controls at all. I would love to see what my app looks like with a pure Uno Material control set.

When I created the solution, the Material option was selected. I see in the documentation that a namespace attribute is being added to the Page element: `<Page xmlns:ut="using:Uno.Themes">`. Since I am using C# Markup, I don't know how to apply this element to the page object. I am wondering if that is causing my controls to look different.

**Comment by kazo0** [comment]
Hi @mobynet1!

It looks like the style of TextBox that you want to be using is the Filled style, as opposed to the default Outlined style that you are seeing.

You can find a listing of all available styles in the [docs here](https://platform.uno/docs/articles/external/uno.themes/doc/material-controls-styles.html)

![image](https://github.com/user-attachments/assets/dac3fe3d-e22c-4a00-8be6-6283c55cae05)

There are also helper classes in the Themes/Material C# Markup packages that make it cleaner to resolve the proper style keys. You can see how those are used in the example code below where I am setting the Style of the TextBoxes

```csharp
public MainPage()
{
    this
        .Background(Theme.Brushes.Background.Default)
        .Content(new StackPanel()
        .VerticalAlignment(VerticalAlignment.Center)
        .HorizontalAlignment(HorizontalAlignment.Center)
        .Spacing(10)
        .Children(
            new TextBlock()
                .Text("Outlined:"),
            new TextBox()
                .Width(300)
                .Header("First Name")
                .Style(Theme.TextBox.Styles.Outlined),
            new TextBlock()
                .Text("Filled:"),
            new TextBox()
                .Width(300)
                .Header("First Name")
                .Style(Theme.TextBox.Styles.Filled)
        ));
}
```

https://github.com/user-attachments/assets/b0a6430c-7a57-4289-99b0-56d97011b39f


I have attached the .zip of the sample app that I used for the above example:

[UnoApp65.zip](https://github.com/user-attachments/files/16457217/UnoApp65.zip)


Let me know if you need any more help with this!

**Comment by francoistanguay** [comment]
Interesting. I just createad a Blank Template + used C# Markup instead of XAML, checked Material Theme, changed the default code from new TextBlock() to new TextBox() and I get a Material-Themed TextBox as expected

![image](https://github.com/user-attachments/assets/0c0df4ca-892c-4493-abf6-53cf961672e0)

In App.xaml.cs default Material+C# Markup template, you should see Material Theme bootstrapped:

![image](https://github.com/user-attachments/assets/67b143db-000e-43fd-84fe-c61118bb22b5)

**Reply by francoistanguay** [reply]
Sorry, missed Steve's answer while typing mine.

**Comment by mobynet1** [comment, marked as answer]
@kazo0, @francoistanguay After I posted this, I realized that I was indeed using the Material controls. And, @kazo0, the styles you added to the textbox made all sorts of lights go on in my head!! I had been reading that very page you posted just hours ago, but it didn't dawn in me that this is how they were to be applied. THANKS!!!!!!!!!!!

I am finding the Uno community to be filled with helpful people with great advice. I also like the fact that I am being heard... This is something that is definitely missing in the Maui community, sadly.

Thanks to everyone!
Steve

**Comment by sasakrsmanovic** [comment]
@mobynet1  Glad to hear! :) ...Close it as answered then ? :)
